homeassistant/components/indevolt/config_flow.py

```diff
@@ -0,0 +1,222 @@
+"""Config flow for Indevolt integration."""
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.service_info.zeroconf import ZeroconfServiceInfo
+
+from .const import DEFAULT_PORT, DEFAULT_SCAN_INTERVAL, DOMAIN, SUPPORTED_MODELS
+from .indevolt import Indevolt
+from .utils import get_device_gen
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class IndevoltConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Indevolt."""
+
+    VERSION = 1
+
+    host = ""
+    port: int
+    model = ""
+    device_sn = ""
+    fw_version = ""
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
+        """Handle the initial step via user interface."""
+        errors = {}
+        if user_input is not None:
+            self.host = user_input["host"]
+            self.port = user_input.get("port", DEFAULT_PORT)
```

Comment:
There's a default already in the data form schema.
```suggestion
            self.port = user_input["port"]
```